### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by zakiyousufi**
### **In this pull request I have added some changes to my database according to the requirements:**

- I have added a column species of type string to my animals table. and modified my schema.sql file.
- I have inserted the required data into my database.
- I added these insertions to my data.sql file.
- Inside a transaction, I updated the animals table by setting the species column to unspecified. and verified that change was made. Then I rolled back the change and verify that the species columns went back to the state before the transaction.
- Inside a transaction:
- I have updated the animals table by setting the species column to digimon for all animals that have a name ending on mon.
- I have updated the animals table by setting the species column to pokemon for all animals that don't have species already set.
- I commited the transaction.
- I verified that change was made and persists after commit.
- Inside a transaction I deleted all records in the animals table, then rolled back the transaction.
- After the rollback I verified if all records in the animals table still exists.
- Inside a transaction:
- I deleteed all animals born after Jan 1st, 2022.
- I created a savepoint for the transaction.
- I updated all animals' weight to be their weight multiplied by -1.
- I rolledback to the savepoint
- I updated all animals' weights that are negative to be their weight multiplied by -1.
- I have committed the transaction
- I write queries to answer the following questions:
- How many animals are there?
- How many animals have never tried to escape?
- What is the average weight of animals?
- Who escapes the most, neutered or not neutered animals?
- What is the minimum and maximum weight of each type of animal?
- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
- I have added these queries to my queries.sql file.

**_I have provided some screenshots of my actions:_**

s1
s2
s3
s4
s5
s6
s7

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -13,4 +13,32 @@ INSERT 0 1
...
+INSERT 0 1
+
+vet_clinic=# INSERT INTO animals (ID,Name,Date_of_birth,Weight_kg,Neutered,Escape_attempts)
+VALUES (8, 'Angemon', 'Jun/12/2005', -45, True, 1);
+INSERT 0 1
+
+vet_clinic=# INSERT INTO animals (ID,Name,Date_of_birth,Weight_kg,Neutered,Escape_attempts)
+VALUES (9, 'Boarmon', 'Jun/7/2005', 20, True, 7);
+INSERT 0 1
+
```
Comment (1GORDON) [inline data.sql]: Great job @zakiyousufi 🤝

- [ ] Kindly make all your `weight_kg` values floating numbers as they are provided in the project requirements.

File: queries.sql [hunk 2]
```diff
@@ -58,3 +58,11 @@ vet_clinic=# SELECT * FROM animals WHERE (weight_kg>= 10.4 AND  weight_kg<= 17.3
...
   3 |         | 2021-01-07    |               1 | f        |     15.04
 (3 rows)
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT MIN(weight_kg) FROM animals;
+SELECT MAX(weight_kg) FROM animals;
+SELECT AVG(escape_attempts) FROM animals GROUP BY date_of_birth BETWEEN '1990-01-01' and '2000-12-31';
```
Comment (1GORDON) [inline queries.sql]: Great job @zakiyousufi 🤝

- [ ] Inside a transaction, update the `animals` table by setting the species column to `unspecified`. Verify that the change was made. Then roll back the change and verify that the `species` columns went back to the state before the transaction. Kindly add the screenshot of this requirement to your Pull Request.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @zakiyousufi ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (peterrobert) [reply data.sql]: This change has been done well

File: queries.sql (hunk 2 above)
Comment (peterrobert) [reply queries.sql]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @zakiyousufi, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**